### ISSUE TITLE
CA1801 is reported for event-handler method

### ISSUE THREAD

**Issue by Tiergan** [dotnet/roslyn-analyzers#2593]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3

#### Diagnostic ID

[CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801-review-unused-parameters)

#### Repro steps

1. Create a new UWP project
2. Add button to MainPage.xaml
3. Add event handler for button click
4. Build project and change show issue generated to Build Only

Example:

**XAML**

```csharp
<Page
    x:Class="App.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
        <Button Click="Button_Click"/>
    </Grid>
</Page>
```

**CS**

```csharp
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
        }
        private void Button_Click(object sender, RoutedEventArgs e)
        {
        }
    }
```

#### Expected behavior
CA1801 is not reported as the method is event handler.

#### Actual behavior
Similar to #2458, CA1801 is being reported. Although the sub type is not EventArgs, the intention is the same and should be ignored. It is also impossible to change RoutedEventArgs to EventArgs to mitigate this issue.

**Comment by sharwell** [comment]
Related to #2338 and #2458

**Comment by sharwell** [comment]
@mavasani Could the exclusion for `EventArgs`-derived types be failing because this is a UWP project?

**Comment by mavasani** [comment]
Yes, this is indeed correct. I don't believe the analyzer should be adjusted to bail out on this UWP specific issue. I believe there are 2 possible ways to address this:
1. Users should mark parameters that are required just for signature constraint as discard parameter named `_`. I am also going to enhance the analyzer to treat names such as `_1`, `_2` as discard symbol names, so multiple parameters can be made discards.
2. The UWP library owners can author a [DiagnosticSuppressor](https://github.com/dotnet/roslyn/pull/36067) that detects this case and suppresses CA1801. The suppressor should ship along with the framework/library as a development dependency.

I am going to mark this issue as external/by-design.
